File: Backend/Backend/Models/Shelters/Address.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models.Shelters
{
    public class Address
    {
        [Required]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string City { get; set; }

        [Required]
        [MaxLength(100)]
        public string Street { get; set; }

        [Required]
        [MaxLength(10)]
        public string PostCode { get; set; }

        [Required]
        [MaxLength(20)]
        public string BuildingNumber { get; set; }

        [MaxLength(200)]
        public string AdditionalAddressLine { get; set; }
    }
}

File: Frontend/FrontendApp/src/app/models/address.ts
export class Address {
    city: string;
    street: string;
    postCode: string;
    buildingNumber: string;
    additionalAddressLine?: string;

    constructor(city: string, street: string, postCode: string, buildingNumber: string, additionalAddressLine: string = '') {
        this.city = city;
        this.street = street;
        this.postCode = postCode;
        this.buildingNumber = buildingNumber;

        if (additionalAddressLine !== '') {
            this.additionalAddressLine = additionalAddressLine;
        }
    }
}